python/ray/util/dask/scheduler_utils.py

```diff
@@ -0,0 +1,373 @@
+import os
+from queue import Queue, Empty
+
+from dask import config
+from dask.callbacks import local_callbacks, unpack_callbacks
+from dask.core import (
+    _execute_task, flatten, get_dependencies, has_tasks, reverse_dict)
+from dask.order import order
```

Comment:
Note that we're still relying on internal Dask APIs, but these APIs are much more narrow and much more stable.